lib/Extension/PhpCodeSniffer/Model/PhpCodeSnifferProcess.php

```diff
@@ -124,4 +145,14 @@ private function runDiagnosticts(TextDocumentItem $textDocument, array $options
             return $stdout;
         });
     }
+
+    private function createTempFile(string $text): string
+    {
+        $tmpName = tempnam(sys_get_temp_dir(), 'phpcsls');
+        $name = sprintf('%s.php', $tmpName);
+        rename($tmpName, $name);
+        file_put_contents($name, $text);
```

Comment:
probably need some error handling here, I think any one of these operations can fail